event-schema/event_schema.go

```diff
@@ -628,138 +667,121 @@ func (manager *EventSchemaManagerT) flushEventSchemas() {
 		}
 	}()
 
-	// This will run forever. If you want to quit in between, change it to ticker and call stop()
-	// Otherwise the ticker won't be GC'ed
-	ticker := time.Tick(flushInterval)
-	for range ticker {
-		if !areEventSchemasPopulated {
-			continue
-		}
+	// If needed, copy the maps and release the lock immediately
+	manager.eventModelLock.Lock()
+	manager.schemaVersionLock.Lock()
 
-		// If needed, copy the maps and release the lock immediately
-		manager.eventModelLock.Lock()
-		manager.schemaVersionLock.Lock()
+	schemaVersionsInCache := make([]*SchemaVersionT, 0)
+	for _, sv := range updatedSchemaVersions {
+		schemaVersionsInCache = append(schemaVersionsInCache, sv)
+	}
 
-		schemaVersionsInCache := make([]*SchemaVersionT, 0)
-		for _, sv := range updatedSchemaVersions {
-			schemaVersionsInCache = append(schemaVersionsInCache, sv)
-		}
+	if len(updatedEventModels) == 0 && len(schemaVersionsInCache) == 0 {
+		manager.eventModelLock.Unlock()
+		manager.schemaVersionLock.Unlock()
+		return nil
+	}
 
-		if len(updatedEventModels) == 0 && len(schemaVersionsInCache) == 0 {
-			manager.eventModelLock.Unlock()
-			manager.schemaVersionLock.Unlock()
-			continue
-		}
+	flushDBHandle = createDBConnection()
 
-		flushDBHandle = createDBConnection()
+	txn, err := flushDBHandle.Begin()
+	assertError(err)
```

Comment:
A panic will shut the whole thing down and it could happen easily since in this case maybe the DB is just not responding. I also see that `flushEventSchemasToDB` actually returns an `error` so why not returning and removing the `recover` on `defer`?

Another question, if we cannot flush, would it make sense to retry or we actually want to throw away everything and restart from scratch? Can you explain a bit the requirements here please?